lib/nfsbase.pm

```diff
@@ -0,0 +1,146 @@
+package nfsbase;
+use mmapi;
+use testapi;
+use strict;
+use warnings;
+use utils;
+use mm_network;
+use version_utils;
+
+
+our @ISA    = qw(Exporter);
+our @EXPORT = qw(server_configure_network try_nfsv2 prepare_exports yast_handle_firewall add_rw_share mount_export client_common_tests);
+
+sub server_configure_network {
+    my ($self) = @_;
+
+    # Configure static IP for client/server test
+    configure_default_gateway;
+    configure_static_ip('10.0.2.101/24');
+    configure_static_dns(get_host_resolv_conf());
+
+    if (is_sle('15+')) {
+        record_soft_failure 'boo#1130093 No firewalld service for nfs-kernel-server';
+        systemctl 'stop ' . $self->firewall;
+        systemctl 'disable ' . $self->firewall;
+    }
+}
+
+sub try_nfsv2 {
+    # Try that NFSv2 is disabled by default
+    systemctl 'restart nfsserver';
+    assert_script_run "cat /proc/fs/nfsd/versions | grep '\\-2'";
+    assert_script_run "sed -i 's/MOUNTD_OPTIONS=.*/MOUNTD_OPTIONS=\"-V2\"/g' /etc/sysconfig/nfs";
+    assert_script_run "sed -i 's/NFSD_OPTIONS=.*/NFSD_OPTIONS=\"-V2\"/g' /etc/sysconfig/nfs";
+    systemctl 'restart nfsserver';
+    assert_script_run "cat /proc/fs/nfsd/versions | grep '+2'";
+
+    # Disable NFSv2 again
+    assert_script_run "sed -i 's/MOUNTD_OPTIONS=.*/MOUNTD_OPTIONS=\"\"/g' /etc/sysconfig/nfs";
+    assert_script_run "sed -i 's/NFSD_OPTIONS=.*/NFSD_OPTIONS=\"\"/g' /etc/sysconfig/nfs";
+    systemctl 'stop nfsserver';
+}
+
+sub prepare_exports {
+    # Create a directory and place a test file in it
+    assert_script_run 'mkdir /srv/nfs && echo success > /srv/nfs/file.txt';
+
+    # Create also hardlink, symlink and do bindmount
+    assert_script_run '( mkdir /srv/dir /srv/nfs/bindmounteddir && ln -s /srv/dir /srv/nfs/symlinkeddir && mount --bind /srv/dir /srv/nfs/bindmounteddir )';
+    assert_script_run '( echo example > /srv/dir/example && ln /srv/dir/example /srv/nfs/hardlinkedfile & ln -s /srv/dir/example /srv/nfs/symlinkedfile )';
+    assert_script_run 'echo secret > /srv/nfs/secret.txt && chmod 740 /srv/nfs/secret.txt';
+
+    # Create large file and count its md5sum
+    assert_script_run 'fallocate -l 1G /srv/nfs/random';
+    assert_script_run "md5sum /srv/nfs/random | cut -d' ' -f1 > /srv/nfs/random.md5sum";
+}
+
+sub yast_handle_firewall {
+    if (is_sle('<15')) {
+        send_key 'alt-f';    # Open port in firewall
+        assert_screen 'nfs-firewall-open';
+    }
+    else {
+        sleep 1;
+        save_screenshot;
+    }
+}
+
+sub add_rw_share {
+    # Add rw share
+    send_key 'alt-d';
+    assert_screen 'nfs-new-share';
+    type_string '/srv/nfs';
+    send_key 'alt-o';
+
+    # Permissions dialog
+    assert_screen 'nfs-share-host';
+    send_key 'tab';
+    # Change 'ro,root_squash' to 'rw,fsid=0,no_root_squash,...'
+    send_key 'home';
+    send_key 'delete';
+    send_key 'delete';
+    send_key 'delete';
+    type_string "rw,fsid=0,no_";
+    send_key 'alt-o';
+
+    # Saved
+    sleep 1;
```

Comment:
default timeout of assert_screen is 30 seconds if it is not enough you can increase it by providing second parameter to assert_screen . Unconditional sleeps is cancer of automation try to avoid them as much as possible . and there is always the way how to avoid them